components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Extensions/StackTypesEx.cs

```diff
@@ -637,26 +563,30 @@ public static class StackTypesEx {
         /// Get network message content mask
         /// </summary>
         /// <param name="mask"></param>
+        /// <param name="fieldMask"></param>
         /// <param name="encoding"></param>
         /// <returns></returns>
-        public static uint ToStackType(this DataSetContentMask? mask, MessageEncoding? encoding) {
+        public static uint ToStackType(this DataSetContentMask? mask,
+            DataSetFieldContentMask? fieldMask, MessageEncoding? encoding) {
             if (mask == null) {
                 mask =
                     DataSetContentMask.DataSetWriterId |
```

Comment:
does this mean we encode both DataSetWriterId and DataSetWriterName? I thought we only use the latter.

Reply:
Yes, both are encoded, but we can update the Json MessageProfile(s) to remove the DataSetWriterId flag.  Then we need to also update the encoder to treat DataSetWriterName flag as DataSetWriterId.  Let me think about this a bit.  What should we do?

Reply:
Ok, current encoder only writes the DataSetWriterName when in --strict mode. It then also writes the DataSetWriterId value.  When not in --strict mode, only DataSetWriterId with string is written.  I think that is ok, it provides a good level of backwards compatibility.  At the expends of having DataSetWriterId in --strict mode.